### PR TITLE
Minimal reproduction of deepCopy bug

### PR DESCRIPTION
**Pull request by nicktindall**
This is the obscure incantation required to reproduce #683

### CODE REVIEW

File: src/test/java/net/openhft/chronicle/wire/WiresTest.java [hunk 1]
```diff
@@ -265,6 +266,32 @@ public void copyToContainsBytesMarshallable() {
...
+                        "      name: EX1,\n" +
+                        "      val: aaa,\n" +
+                        "   }\n" +
+                        "}\n");
+        final Thing thingCopy = thing2.deepCopy();
+        assertEquals(thing2, thingCopy);
+    }
+
+    static class Thing extends AbstractEventCfg<Thing> {
+        @AsMarshallable
```
Comment (nicktindall) [inline src/test/java/net/openhft/chronicle/wire/WiresTest.java:286]: Needs to be `@AsMarshallable`

File: src/test/java/net/openhft/chronicle/wire/WiresTest.java [hunk 2]
```diff
@@ -265,6 +266,32 @@ public void copyToContainsBytesMarshallable() {
...
+        DynamicEnum dee1;
+    }
+
+    enum EnumThing implements DynamicEnum {
+        ONE,
+        TWO;
```
Comment (nicktindall) [inline src/test/java/net/openhft/chronicle/wire/WiresTest.java:293]: Seems to need to be a DynamicEnum

File: src/test/java/net/openhft/chronicle/wire/WiresTest.java [hunk 3]
```diff
@@ -265,6 +266,31 @@ public void copyToContainsBytesMarshallable() {
         assertEquals(containsBM.inner.name, containsBM2.inner.name);
     }
 
+    @Test
+    public void deepCopyWillWorkWhenDifferentDynamicEnumIsAnnotatedAsMarshallable() {
+        ClassAliasPool.CLASS_ALIASES.addAlias(Thing.class, EnumThing.class);
+
+        Thing thing2 = Marshallable.fromString(
+                "!Thing {" +
+                        "   eventTime: 2020-09-09T01:46:41,\n" +
```
Comment (nicktindall) [inline src/test/java/net/openhft/chronicle/wire/WiresTest.java:275]: Needs to be another field in the DTO

**Comment by peter-lawrey** [comment]
Assigned to me as this only reproduces the issue, and doesn't have a fix.

**Comment by nicktindall** [comment]
> Assigned to me as this only reproduces the issue, and doesn't have a fix.

@peter-lawrey the fix is here https://github.com/OpenHFT/Chronicle-Wire/pull/681

**Comment by nicktindall** [comment]
Closed as the test is now included on #727

**Pull request closed without merging**